include/rdm/param_data.h

```diff
@@ -0,0 +1,585 @@
+typedef struct RdmPdScopeString
+{
+  char string[RDMPD_MAX_SCOPE_STR_LEN]; /*!< The string data. */
+} RdmPdScopeString;
+
+typedef struct RdmPdComponentScope
+{
+  uint16_t scope_slot;
+  RdmPdScopeString scope_string;
+  EtcPalSockaddr static_broker_saddr;
+} RdmPdComponentScope;
+
+etcpal_error_t rdmpd_unpack_get_resp_component_scope(const RdmParamData *pd, RdmPdComponentScope *scope_out);
+etcpal_error_t rdmpd_pack_get_resp_component_scope(const RdmPdComponentScope *scope, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_component_scope(const RdmParamData *pd, RdmPdComponentScope *scope_out);
+etcpal_error_t rdmpd_pack_set_component_scope(const RdmPdComponentScope *scope, RdmParamData *pd_out);
+
+// Get/Set SEARCH_DOMAIN
+/*! The maximum length of a search domain string, including one extra byte for a null terminator. */
+#define RDMPD_MAX_SEARCH_DOMAIN_STR_LEN 232
+
+/*! A structure representing a search domain string. */
+typedef struct RdmPdSearchDomain
+{
+  char string[RDMPD_MAX_SEARCH_DOMAIN_STR_LEN]; /*!< The string data. */
+} RdmPdSearchDomain;
+
+etcpal_error_t rdmpd_unpack_get_resp_search_domain(const RdmParamData *pd, RdmPdSearchDomain *domain_name_out);
+etcpal_error_t rdmpd_pack_get_resp_search_domain(const RdmPdSearchDomain *domain_name, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_search_domain(const RdmParamData *pd, RdmPdSearchDomain *domain_name_out);
+etcpal_error_t rdmpd_pack_set_search_domain(const RdmPdSearchDomain *domain_name, RdmParamData *pd_out);
+
+// Get/Set TCP_COMMS_STATUS
+typedef struct RdmPdTcpCommsEntry
+{
+  RdmPdScopeString scope_string;
+  EtcPalSockaddr broker_saddr;
+  uint16_t unhealthy_tcp_events;
+} RdmPdTcpCommsEntry;
+
+etcpal_error_t rdmpd_unpack_get_resp_tcp_comms_status(const RdmParamData *pd, RdmPdTcpCommsEntry *entry_out);
+etcpal_error_t rdmpd_pack_get_resp_tcp_comms_status(const RdmPdTcpCommsEntry *entry, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_tcp_comms_status(const RdmParamData *pd, RdmPdScopeString *scope_out);
+etcpal_error_t rdmpd_pack_set_tcp_comms_status(const RdmPdScopeString *scope, RdmParamData *pd_out);
+
+// Get/Set BROKER_STATUS
+typedef enum
+{
+  kRdmPdBrokerStateDisabled = 0x00,
+  kRdmPdBrokerStateActive = 0x01,
+  kRdmPdBrokerStateStandby = 0x02
+} rdmpd_broker_state_t;
+
+typedef struct RdmPdBrokerStatus
+{
+  bool set_allowed;
+  rdmpd_broker_state_t broker_state;
+} RdmPdBrokerStatus;
+
+etcpal_error_t rdmpd_unpack_get_resp_broker_status(const RdmParamData *pd, RdmPdBrokerStatus *status_out);
+etcpal_error_t rdmpd_pack_get_resp_broker_status(const RdmPdBrokerStatus *status, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_broker_status(const RdmParamData *pd, uint8_t *broker_state_out);
+etcpal_error_t rdmpd_pack_set_broker_status(uint8_t broker_state, RdmParamData *pd_out);
+
+/* e1.37-7 Table A-1: RDM Parameter ID Defines */
+// Get ENDPOINT_LIST
+typedef enum
+{
+  kRdmPdEndptTypeVirtual = 0x00,
```

Comment:
Per similar comment - just a gut feeling that `Endpt` should be expanded to `Endpoint` here.

Reply:
Done.